app/src/main/java/de/tum/in/tca/ticketcheck/component/ticket/EventsController.java

```diff
@@ -34,115 +32,47 @@
     public EventsController(Context context) {
         this.context = context;
         eventDao = TcaDb.getInstance(context).eventDao();
-        ticketDao = TcaDb.getInstance(context).ticketDao();
-        ticketTypeDao = TcaDb.getInstance(context).ticketTypeDao();
+        downloadFromService();
     }
 
 
-    public void downloadFromService(boolean force) {
-        TUMCabeClient api = TUMCabeClient.getInstance(context);
-
-        // Delete all too old items
-        eventDao.cleanUp();
-
-        // Load all events since the last sync
-        try {
-            List<Event> events = api.getEvents();
-            eventDao.insert(events);
-        } catch (IOException e) {
-            Utils.log(e);
-        }
-
-        // Load all tickets
-        try {
-            if(Utils.getSetting(context, Const.CHAT_MEMBER, ChatMember.class) != null) {
-                api.getTickets(context, new Callback<List<Ticket>>() {
-                    @Override
-                    public void onResponse(Call<List<Ticket>> call, Response<List<Ticket>> response) {
-                        List<Ticket> list = response.body();
-                        if (list == null) list = new ArrayList<Ticket>();
-                        ticketDao.insert(list);
-                    }
-
-                    @Override
-                    public void onFailure(Call<List<Ticket>> call, Throwable t) {
-                        //TODO: inform user about failure
-                        t.printStackTrace();
-                    }
-                });
+    private void downloadFromService() {
+        // get event information from API
+        Callback<List<Event>> cb = new Callback<List<Event>>() {
+            @Override
+            public void onResponse(Call<List<Event>> call, Response<List<Event>> response) {
+                // set view
+                List<Event> events = response.body();
+                eventDao.insert(events);
             }
-        } catch (IOException e) {
-            Utils.log(e);
-        }
 
-        // Load all ticket types
+            @Override
+            public void onFailure(Call<List<Event>> call, Throwable t) {
+                t.printStackTrace();
+                // try to connect again
+                downloadFromService();
```

Comment:
possible endless loop if offline. Please use `Utils.log` instead of printStackTrace.

Reply:
Removed retry, user should refresh manually if the download fails